### ISSUE TITLE
Not recognized as markdown formatter in Jupyter Files in VS Code

### ISSUE THREAD

**Issue by vacuum136** [yzhang-gh/vscode-markdown#1095]
## Problem

Can't be recognized as markdown formatter in the markdown cell embedded in VS Code Jupyter Files (.ipynb). In normal markdown file, formatting works without any problem.

![image](https://user-images.githubusercontent.com/17034840/159144210-87f18798-f220-4b74-9a47-f5043994c4af.png)

**Comment by yzhang-gh** [comment]
Could you check whether other Markdown formatters (e.g. Markdownlint, Prettier) work well in `.ipynb` fils?

**Comment by Lemmingh** [comment]
<https://github.com/microsoft/vscode/issues/141717>

The `Uri` scheme of a notebook cell is none of them:

<https://github.com/yzhang-gh/vscode-markdown/blob/fe80c74e8aa2ad5e6f1b03aaac29f209cb4fbebb/src/util/generic.ts#L4-L8>

**Comment by Lemmingh** [comment]
Besides, `DocumentFilter` will have conditional semantics in VS Code 1.66:

<https://github.com/microsoft/vscode/issues/141143#issuecomment-1067015245>

---

We need to revisit our [usage of document selectors](https://code.visualstudio.com/api/references/document-selector#summary).

**Comment by vacuum136** [comment]
> Could you check whether other Markdown formatters (e.g. Markdownlint, Prettier) work well in `.ipynb` files?

Yes, I checked one Markdown formatter (Markdown Table Prettifier), it works.

**Comment by Lemmingh** [comment]
Make sense. Its selector is just a simple `"markdown"`:

<https://github.com/darkriszty/MarkdownTablePrettify-VSCodeExt/blob/3523a95fb07829b3a550b50e34c7c9c745b9ac29/src/extension/extension.ts#L13>

Is it safe for us to use such a loose selector here? @yzhang-gh

**Comment by yzhang-gh** [comment]
> Is it safe for us to use such a loose selector here?

I think so as I can't think of any side effects for now.
